scanpy/plotting/_tools/__init__.py

```diff
@@ -382,26 +388,41 @@ def _rank_genes_groups_plot(
         groupby = str(adata.uns[key]['params']['groupby'])
     group_names = adata.uns[key]['names'].dtype.names if groups is None else groups
 
-    var_names = {}  # dict in which each group is the key and the n_genes are the values
-    gene_names = []
-    for group in group_names:
-        if min_logfoldchange is not None:
-            df = rank_genes_groups_df(adata, group, key=key)
-            # select genes with given log_fold change
-            genes_list = df[df.logfoldchanges > min_logfoldchange].names.tolist()[
-                :n_genes
-            ]
-        else:
-            # get all genes that are 'non-nan'
-            genes_list = [
-                gene for gene in adata.uns[key]['names'][group] if not pd.isnull(gene)
-            ][:n_genes]
+    gene_symbols = kwds.get('gene_symbols', None)
 
-        if len(genes_list) == 0:
-            logg.warning(f'No genes found for group {group}')
-            continue
-        var_names[group] = genes_list
-        gene_names.extend(genes_list)
+    if gene_names is not None:
+        var_names = gene_names
+        if isinstance(var_names, Mapping):
+            # get a single list of all gene names in the dictionary
+            gene_names = sum([list(x) for x in var_names.values()], [])
+        elif isinstance(var_names, str):
+            gene_names = [var_names]
+        else:
+            gene_names = var_names
+    else:
+        # dict in which each group is the key and the n_genes are the values
+        var_names = {}
+        gene_names = []
+        for group in group_names:
+            df = rank_genes_groups_df(adata, group, key=key, gene_symbols=gene_symbols)
+            if min_logfoldchange is not None:
+                # select genes with given log_fold change
+                df = df[df.logfoldchanges > min_logfoldchange]
+
+            if gene_symbols is not None:
+                df['names'] = df['symbol']
```

Comment:
`"symbol"` shouldn't be hardcoded, as it's not hardcoded in the output of `sc.get.rank_genes_groups`

```suggestion
                df['names'] = df[gene_symbols]
```

Reply:
Solved